lib/src/component/error_boundary_recoverable.dart

```diff
@@ -9,15 +14,224 @@ part 'error_boundary_recoverable.over_react.g.dart';
+  //
+  // [2] Otherwise, we can't currently make any assumptions about what the consumer would like to have
+  //     rendered as a fallback when a component throws - so we just remount the children
+  //     as a result of keeping `state.showFallbackUIOnError` set to false.
+  //
+  //     [2.1] However, as a way to handle the edge case where the exact same component wrapped by an
+  //           `ErrorBoundary` throws the exact same error more than once before the `_identicalErrorTimer`
+  //           "times out", we should not just keep remounting since the component
+  //           should - at that point - be considered "unrecoverable",
+  //           and the repeated errors will most likely lock up the main thread.
+  //
+  //     [2.2] So we'll set `state.showFallbackUIOnError` to true, which will cause our default fallback UI
+  //           to be rendered - which is simply a String snapshot of the DOM (`_domAtTimeOfError`).
+  //
+  //           [2.2.1] We will also fire `props.onComponentIsUnrecoverable` if set - to give the
+  //                   consumer the ability to have knowledge of the unrecoverable error state - and
+  //                   to optionally manually recover from the error by:
+  //
+  //                   1. Passing in a new child (see: componentDidUpdate)
+  //                   2. Calling the `reset()` instance method
+  //
+  //           [2.2.2] Since we should __never__ throw an error from our... uh... error boundary,
+  //                   wrap in a try catch just in case `findDomNode` throws as a result of the
+  //                   wrapped react tree rendering a string instead of a composite or dom component.
+  //
+  // [3] Log the caught error using a logger if `props.shouldLogErrors` is true.
+  // ---------------------------------------------- /\ ----------------------------------------------
+
+  String _domAtTimeOfError;
+  List<String> _errorLog = [];
+  List<ReactErrorInfo> _callStackLog = [];
+  Timer _identicalErrorTimer;
+
+  /// Called by [componentDidCatch].
+  void _handleErrorInComponentTree(/*Error||Exception*/dynamic error, ReactErrorInfo info) {
+    // ----- [1] ----- //
+    if (props.fallbackUIRenderer != null) {
+      _errorLog.add(error.toString());
+      _callStackLog.add(info);
+      _logErrorCaughtByErrorBoundary(error, info); // [3]
+      return;
+    }
+    // ----- [2] ----- //
+    else {
+      bool sameErrorWasThrownTwiceConsecutively = false;
+      final errorString = error.toString();
+
+      for (var i = 0; i < _errorLog.length; i++) {
+        if (_errorLog[i] == errorString && _callStackLog[i].componentStack == info.componentStack) {
+          sameErrorWasThrownTwiceConsecutively = true;
+          break;
+        }
+      }
+
+      if (sameErrorWasThrownTwiceConsecutively) { // [2.1]
+        try { // [2.2.2]
+          _domAtTimeOfError = findDomNode(this)?.innerHtml; // [2.2]
```

Comment:
This is a copy of the code here: https://github.com/Workiva/over_react/blob/8a52923b0297565f31160f7bff1c258c202f5bae/lib/src/component/_deprecated/error_boundary_recoverable.dart#L162